sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -110,6 +111,7 @@ object Literal {
     case _ if clz == classOf[Timestamp] => TimestampType
     case _ if clz == classOf[JavaBigDecimal] => DecimalType.SYSTEM_DEFAULT
     case _ if clz == classOf[Array[Byte]] => BinaryType
+    case _ if clz == classOf[Array[Char]] => StringType
```

Comment:
Should we need to consider `java.lang.Character`?

Reply:
It seems we can never go into here with `java.lang.Character`. This method is used which data type is `Array[_]`, but we always intercept the `Array[Char]` first.

Reply:
Hm, but looks like `JavaByte ` is being handled here. It has `Array[Byte]` pattern here too.

Reply:
I'm ok to follow the behavior. Add this pattern.